javaslang/src/main/java/javaslang/collection/JavaConverters.java

```diff
@@ -0,0 +1,158 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import java.util.Collection;
+import java.util.ListIterator;
+
+/**
+ * Internal class, containing helpers.
+ *
+ * @author Pap Lőrinc
+ * @since 2.1.0
+ */
+final class JavaConverters {
+    @SuppressWarnings("unchecked")
+    static class SeqList<S extends Seq<T>, T> implements java.util.List<T> {
+        protected final S delegate;
+
+        SeqList(S delegate) {
+            this.delegate = delegate;
+        }
+
+        @Override
+        public int size() {
+            return delegate.size();
+        }
+
+        @Override
+        public boolean isEmpty() {
+            return delegate.isEmpty();
+        }
+
+        @Override
+        public boolean contains(Object element) {
+            return delegate.contains((T) element);
+        }
+
+        @Override
+        public boolean containsAll(Collection<?> elements) {
+            final Collection<? extends T> collection = (Collection<? extends T>) elements;
+            return delegate.containsAll(Iterator.ofAll(collection.iterator()));
+        }
+
+        @Override
+        public java.util.Iterator<T> iterator() {
+            return delegate.iterator();
+        }
+
+        @Override
+        public T get(int index) {
+            return delegate.get(index);
+        }
+
+        @Override
+        public int indexOf(Object element) {
+            return delegate.indexOf((T) element);
+        }
+
+        @Override
+        public int lastIndexOf(Object element) {
+            return delegate.lastIndexOf((T) element);
+        }
+
+        @Override
+        public java.util.List<T> subList(int from, int to) {
+            return delegate.slice(from, to).asJava();
```

Comment:
**Update:** List.subList(int, int) does throw IndexOutOfBoundsException, so should we. slice() does not throw, so we have to use subSequence().